lib/authCodeGrant.js

```diff
@@ -172,7 +173,11 @@ function generateCode (done) {
  */
 function saveAuthCode (done) {
   var expires = new Date();
-  expires.setSeconds(expires.getSeconds() + this.config.authCodeLifetime);
+
+  var ttl = !isNaN(parseFloat(this.client.authCodeLifetime)) && isFinite(this.client.authCodeLifetime) ?
+    this.client.authCodeLifetime : this.config.authCodeLifetime;
+
+  expires.setSeconds(expires.getSeconds() + ttl);
```

Comment:
Sure, I think that makes sense, I'll add it once I get an answer to the other two comment questions.